src/configAPIClient.ts

```diff
@@ -91,78 +90,95 @@ export interface IConfigResponse {
+        : new XHRUploader(uploadUrl);
+
+    this.getSDKConfiguration = async (): Promise<IConfigResponse> => {
+        let configResponse: IConfigResponse;
+        const fetchPayload: fetchPayload = {
+            method: 'get',
+            headers: {
+                Accept: 'text/plain;charset=UTF-8',
+                'Content-Type': 'text/plain;charset=UTF-8',
+            },
+            body: null,
+        };
+
         try {
-            const xhrCallback = function() {
-                if (xhr.readyState === 4) {
-                    // when a 200 returns, merge current config with what comes back from config, prioritizing user inputted config
-                    if (xhr.status === 200) {
-                        config = mpInstance._Helpers.extend(
-                            {},
-                            config,
-                            JSON.parse(xhr.responseText)
-                        );
-                        completeSDKInitialization(apiKey, config, mpInstance);
-                        mpInstance.Logger.verbose(
-                            'Successfully received configuration from server'
-                        );
-                    } else {
-                        // if for some reason a 200 doesn't return, then we initialize with the just the passed through config
-                        completeSDKInitialization(apiKey, config, mpInstance);
-                        mpInstance.Logger.verbose(
-                            'Issue with receiving configuration from server, received HTTP Code of ' +
-                                xhr.status
-                        );
-                    }
-                }
-            };
-
-            const xhr = mpInstance._Helpers.createXHR(xhrCallback);
-            url =
-                'https://' +
-                mpInstance._Store.SDKConfig.configUrl +
-                apiKey +
-                '/config?env=';
-            if (config.isDevelopmentMode) {
-                url = url + '1';
-            } else {
-                url = url + '0';
-            }
+            const response = await uploader.upload(fetchPayload);
+            if (response.status === 200) {
+                console.log(
+                    'Config API Client - returning successful response'
+                );
+                mpInstance?.Logger?.verbose(
+                    'Successfully received configuration from server'
+                );
 
-            const dataPlan = config.dataPlan as DataPlanConfig;
-            if (dataPlan) {
-                if (dataPlan.planId) {
-                    url = url + '&plan_id=' + dataPlan.planId || '';
-                }
-                if (dataPlan.planVersion) {
-                    url = url + '&plan_version=' + dataPlan.planVersion || '';
+                // https://go.mparticle.com/work/SQDSDKS-6568
+                // FetchUploader returns the response as a JSON object that we have to await
+                if (response.json) {
```

Comment:
is this the only thing that will determine if it is a fetch uploader?  ether way, for readability, can we make the xhrresponse below part of an `else`?  it seems more clear that it's one or the other

Reply:
then return the config response once to DRY it up.  i'm not committed to this idea, just a thought.

Reply:
I'm good with that.